File: ios/ClearCache.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ClearCache, NSObject)

RCT_EXTERN_METHOD(getCacheDirSize:(RCTPromiseResolveBlock)resolve withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(clearCacheDir:(RCTPromiseResolveBlock)resolve withRejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/ClearCache.swift
import Foundation

@objc(ClearCache)
class ClearCache: NSObject {
    
    @objc(getCacheDirSize:withRejecter:)
    func getCacheDirSize(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        let fileManager = FileManager.default
        let cacheDirUrl = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
        
        do {
            let attributes = try fileManager.attributesOfItem(atPath: cacheDirUrl.path)
            resolve(attributes[FileAttributeKey.size])
        } catch {
            reject("event_failure", "no event id returned", nil);
        }
    }
    
    @objc(clearCacheDir:withRejecter:)
    func clearCacheDir(resolve:RCTPromiseResolveBlock, reject:RCTPromiseRejectBlock) -> Void {
        
    }
}
